GW2EIEvtcParser/EIData/Actors/SingleActor.cs

```diff
@@ -519,12 +519,28 @@ public override IEnumerable<CastEvent> GetIntersectingCastEvents(ParsedEvtcLog l
         return CastEvents.Where(x => KeepIntersectingCastLog(x, start, end));
 
     }
+
+    public override IEnumerable<AnimatedCastEvent> GetAnimatedCastEvents(ParsedEvtcLog log, long start, long end)
+    {
+        return log.CombatData.GetAnimatedCastData(AgentItem).Where(x => x.Time >= start && x.Time <= end);
+    }
+
+    public override IEnumerable<InstantCastEvent> GetInstantCastEvents(ParsedEvtcLog log, long start, long end)
+    {
+        return log.CombatData.GetInstantCastData(AgentItem).Where(x => x.Time >= start && x.Time <= end);
+    }
+
+    public override IEnumerable<WeaponSwapEvent> GetWeaponSwapEvents(ParsedEvtcLog log, long start, long end)
```

Comment:
This will not be equivalent to iterating over weapon swaps from GetCastEvents out put, if not necessary, it is better to not have that API.